client/components/AllProducts.js

```diff
@@ -5,39 +5,57 @@ import {
   ListGroup,
   ListGroupItem,
   ButtonGroup,
-  Button
+  Button,
+  Form
 } from 'react-bootstrap'
 import {
   fetchProductsThunk,
   fetchCategoriesThunk,
-  filterProducts
+  filterProductsByCategory,
+  filterProductsByTitle
 } from '../store/product'
 import {Link} from 'react-router-dom'
 
 export class AllProducts extends React.Component {
   constructor(props) {
     super(props)
     this.state = {
-      filterByCategory: ''
+      filterByCategory: '',
+      filterByTitle: ''
     }
     this.handleChange = this.handleChange.bind(this)
+    this.handleSubmit = this.handleSubmit.bind(this)
   }
   componentDidMount() {
     this.props.fetchProducts()
     this.props.fetchCategories()
   }
   handleChange(event) {
-    this.setState({filterByCategory: event}, () => {
-      this.props.filterProducts(this.state.filterByCategory)
+    console.log('event', event.target.value)
+    // this.setState({filterByCategory: event}, () => {
+    //   // I think the setState is going to have to be the following:
```

Comment:
We are on the same page here. I made the same change when working on the fix for category filtering.